pkg/install/v1alpha1/cstor_volume.go

```diff
@@ -621,7 +621,12 @@ spec:
     Calculate the replica count
     Add as many poolUid to resources as there is replica count
     */}}
-    {{- $poolUids := keys .ListItems.cvolPoolList.pools | randomize }}
+    {{- $replicaAffinityName:= .TaskResult.creategetpvc.replicaAntiAffinity -}}
+    {{- if ne $replicaAffinityName  "none" -}}
+    {{- $poolUids := keys .ListItems.cvolPoolList.pools -}}
+    {{- $lblSelector := printf "openebs.io/replica-anti-affinity=%s" .TaskResult.creategetpvc.replicaAntiAffinity -}}
+    {{- $preferAntiAffinity := cspPreferAntiAffinity $lblSelector -}}
+    {{- $poolUids := cspFilter $poolUids $preferAntiAffinity | randomize -}}
```

Comment:
do we need to handle the error thrown, if any, from Filter()?

Reply:
Can we test this